### PR TITLE
Segment fix

### PR DESCRIPTION
**Pull request by jdee**
Addresses #401.

This adds a `react-native-branch-segment` pod that addresses a dependency conflict.

The pods here are only intended for use with hybrid apps: native iOS apps written in Swift or Obj-C that include one or more RN components. If you're using `react-native link` (or `rnpm link`) to build your app, the Branch native iOS SDK is embedded in the `react-native-branch` project.

To use this pod in an app using the Branch Segment integration, add this to your Podfile:

```Ruby
pod "react-native-branch-segment", path: "node_modules/react-native-branch" # or wherever your node_modules is
```

Instead of

```Ruby
pod "react-native-branch", path: "node_modules/react-native-branch"
pod "Branch-SDK", path: "node_modules/react-native-branch/ios"
```

Be sure you're not including the Branch-SDK pod from node_modules. This may not be the last word, but it fixes the conflict and may be considered an officially-supported solution.

### CODE REVIEW

File: react-native-branch-segment.podspec [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  s.authors      = {
+                     'rt2zz' => 'zack@root-two.com',
+                     'Jimmy Dee' => 'jgvdthree@gmail.com'
+                   }
+  s.license      = spec['license']
+  s.homepage     = spec['homepage']
+  s.platform     = :ios, "7.0"
+  s.source       = { spec['repository']['type'].to_sym => spec['repository']['url'].sub(/^[a-z]+\+/, '') }
+  s.source_files = [ "ios/*.h", "ios/*.m"]
+  s.header_dir   = "react-native-branch"
```
Comment (jdee) [inline react-native-branch-segment.podspec]: This ensures that imports don't have to change, if you're importing react-native-branch in your native code. This isn't actually necessary unless you're using `branch.subscribe` in your RN (JS) code. If you're routing links in native code, which seems the most likely thing, you can just use the Branch native iOS SDK directly. This is how webview_example_native_ios is set up. The testbed_native_ios app uses the RNBranch native SDK as an example and verifies that this change builds correctly.

File: react-native-branch-segment.podspec [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  s.dependency 'Branch'
```
Comment (jdee) [inline react-native-branch-segment.podspec]: Corrected dependency

File: react-native-branch-segment.podspec [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# Not for use in pure RN apps.
+
+# expect package.json in current dir
+package_json_filename = File.expand_path("./package.json", __dir__)
+
+# load the spec from package.json
+spec = JSON.load(File.read(package_json_filename))
+
+Pod::Spec.new do |s|
+  s.name             = "#{spec['name']}-segment"
```
Comment (jdee) [inline react-native-branch-segment.podspec]: Pod name change

**Comment by echo-branch** [comment]
Code looks good!  I'll try to test it myself either later tonight or tomorrow.

**Comment by jdee** [comment]
To test, testbed_native_ios is the best option, since it uses the RNBranch native code. Just change the Podfile there as described above. Then in that directory (examples/testbed_native_ios) run:

```bash
yarn
bundle check || bundle install
bundle exec pod install
```

Then you can build the workspace.

**Comment by echo-branch** [comment]
Works in test app!  I see the updated podspec pulls in a single copy of branch.

**Pull request merged**